README.md

```diff
@@ -1 +1,105 @@
-# angular-http-interceptors-readme
\ No newline at end of file
+# HTTP Interceptors
+
+## Overview
+
+## Objectives
+
+- Describe HTTP Interceptors
+- Write an HTTP Interceptor
+
+## What is a HTTP interceptor?
+
+Now, using `$http` or `$resource` means we're going to end up doing a lot of HTTP requests in our applications.
+
+Eventually, we may need to manipulate all of the requests, or even take a wide-look at all the responses and handle any errors globally instead of on an individual case - we can do this with HTTP interceptors.
+
+HTTP interceptors allow us to define functionality that will happen before every request is made, or just after every request has come back from the server. This means we can append things to every request we make, or handle errors that may arise from the backend (it's better to do this in an interceptor and have application-wide error handling, rather than doing it for every possible request we make).
+
+HTTP interceptors are just simple services that we then push into an interceptors array in the `$httpProvider` service.
```

Comment:
When do the interceptors get attached onto the `$httpProvider` service? Is this something that we have to configure?